models/banana_gun/solana/banana_gun_solana_bot_trades.sql

```diff
@@ -0,0 +1,167 @@
+{% set fee_receiver_5 = '4BBNEVRgrxVKv9f7pMNE788XM1tt379X9vNjpDH2KCL7' %}
+{% set wsol_token = 'So11111111111111111111111111111111111111112' %}
+
+WITH
+  allFeePayments AS (
+    SELECT
+      tx_id,
+      'SOL' AS feeTokenType,
+      balance_change / 1e9 AS fee_token_amount,
+      '{{wsol_token}}' AS fee_token_mint_address
+    FROM
+      {{ source('solana','account_activity') }}
+    WHERE
+      {% if is_incremental() %}
+      {{ incremental_predicate('block_time') }}
+      {% else %}
+      block_time >= TIMESTAMP '{{project_start_date}}'
+      {% endif %}
+      AND tx_success
+      AND balance_change > 0
+      AND (
+        address = '{{fee_receiver_1}}'
+        OR address = '{{fee_receiver_2}}'
+        OR address = '{{fee_receiver_3}}'
+        OR address = '{{fee_receiver_4}}'
+        OR address = '{{fee_receiver_5}}'
+      )
+      AND tx_id != 'AT915GhHaLdGsdFkywx2uE6jqSXeyTauveYH2BQqWMyptGhUtjE6dcdr74ErELg79VY9apZ9Egiyc1VtA6Ddykb' -- Edge case that sent fees to multiple fee wallets
+  ),
+  botTrades AS (
+    SELECT
+      trades.block_time,
+      CAST(date_trunc('day', trades.block_time) AS date) AS block_date,
+      CAST(date_trunc('month', trades.block_time) AS date) AS block_month,
+      'solana' AS blockchain,
+      amount_usd,
+      IF(
+        token_sold_mint_address = '{{wsol_token}}',
+        'Buy',
+        'Sell'
+      ) AS type,
+      token_bought_amount,
+      token_bought_symbol,
+      token_bought_mint_address AS token_bought_address,
+      token_sold_amount,
+      token_sold_symbol,
+      token_sold_mint_address AS token_sold_address,
+      fee_token_amount * price AS fee_usd,
+      fee_token_amount,
+      IF(feeTokenType = 'SOL', 'SOL', symbol) AS fee_token_symbol,
+      fee_token_mint_address AS fee_token_address,
+      project,
+      version,
+      token_pair,
+      project_program_id AS project_contract_address,
+      trader_id AS user,
+      trades.tx_id,
+      tx_index,
+      outer_instruction_index,
+      inner_instruction_index
+    FROM
+      {{ ref('dex_solana_trades') }} AS trades
+      JOIN allFeePayments AS feePayments ON trades.tx_id = feePayments.tx_id
+      LEFT JOIN {{ source('prices', 'usd') }} AS feeTokenPrices ON (
+        feeTokenPrices.blockchain = 'solana'
+        AND fee_token_mint_address = toBase58 (feeTokenPrices.contract_address)
+        AND date_trunc('minute', block_time) = minute
+        {% if is_incremental() %}
+        AND {{ incremental_predicate('minute') }}
+        {% else %}
+        AND minute >= TIMESTAMP '{{project_start_date}}'
+        {% endif %}
+      )
+      JOIN {{ source('solana','transactions') }} AS transactions ON (
+        trades.tx_id = id
+        {% if is_incremental() %}
+        AND {{ incremental_predicate('transactions.block_time') }}
+        {% else %}
+        AND transactions.block_time >= TIMESTAMP '{{project_start_date}}'
+        {% endif %}
+      )
```

Comment:
By the way do you think we can make a CTE before inner join to optimize the query like
```
solana_tx as (
    SELECT [ * | neccessary columns ] 
    FROM
    {{ source('solana','transactions') }}
    {% if is_incremental() %}
    WHERE {{ incremental_predicate('transactions.block_time') }}
    {% else %}
    WHERE block_time >= TIMESTAMP '{{project_start_date}}'
    {% endif %}
```